codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/Project.kt

```diff
@@ -97,33 +101,37 @@ class AnalysisProject(
 ) {
 
     /** Analyzes the project and returns the result. */
-    fun analyze(): AnalysisResult {
-        // TODO(oxisto): Replace this mock run object with a real one later on.
-        //  Currently, this is only to show that we support SARIF
+    fun analyze(
+        postProcess: ((TranslationResult) -> Pair<List<ReportingDescriptor>, List<Result>>)? = null
+    ): AnalysisResult {
+        val tr = TranslationManager.builder().config(config).build().analyze().get()
+        val (rules, results) = postProcess?.invoke(tr) ?: Pair(emptyList(), emptyList())
+
+        // Create a new SARIF run, including a tool definition and rules corresponding to the
+        // individual security statements
         val run =
             Run(
-                tool = Tool(driver = ToolComponent(name = "Codyze", version = "x.x.x")),
-                results =
-                    listOf(
-                        Result(
-                            rule = ReportingDescriptorReference(id = "Rule1"),
-                            message = Message(markdown = "This is a **finding**"),
-                            level = Level.Note,
-                            locations = listOf(),
-                        )
-                    ),
+                tool =
+                    Tool(driver = ToolComponent(name = "Codyze", version = "x.x.x", rules = rules)),
+                results = results,
             )
 
-        val result = TranslationManager.builder().config(config).build().analyze().get()
-
-        return AnalysisResult(run = run, translationResult = result)
+        return AnalysisResult(
+            translationResult = tr,
+            sarif = SarifSchema210(version = Version.The210, runs = listOf(run)),
+        )
     }
 
     companion object {
-        /** Builds a translation configuration from the given CLI options. */
-        fun fromOptions(
-            projectOptions: ProjectOptions,
-            translationOptions: TranslationOptions,
+        /** Builds a translation configuration from the given project directory. */
+        fun from(
+            projectDir: Path,
+            sources: List<Path>? = null,
+            components: List<String>? = null,
+            exclusionPatterns: List<String>? = null,
+            configBuilder:
```

Comment:
I find the name of `configModifier` from [TestUtils.kt#L132](https://github.com/Fraunhofer-AISEC/cpg/blob/48bc1a0bf4a5abf9f0f06f81abff742222d9ab30/cpg-core/src/testFixtures/kotlin/de/fraunhofer/aisec/cpg/test/TestUtils.kt#L132) more suitable. We may need it to be more specific and call it `cpgConfigModifier` to differentiate it from the Codyze configuration.

Any thoughts?